disnake/embeds.py

```diff
@@ -794,3 +800,64 @@ def _handle_resource(self, url: Optional[Any], file: File, *, key: _FileKey) ->
         else:
             self._files.pop(key, None)
             return str(url) if url is not None else None
+
+    def check_limits(self):
+        """
+        Checks if this embed fits within the limits dictated by Discord.
+
+        +--------------------------+------------------------------------+
+        |   Field                  |              Limit                 |
+        +--------------------------+------------------------------------+
+        | title                    |        256 characters              |
+        +--------------------------+------------------------------------+
+        | description              |        4096 characters             |
+        +--------------------------+------------------------------------+
+        | fields                   |        Up to 25 field objects      |
+        +--------------------------+------------------------------------+
+        | field.name               |        256 characters              |
+        +--------------------------+------------------------------------+
+        | field.value              |        1024 characters             |
+        +--------------------------+------------------------------------+
+        | footer.text              |        2048 characters             |
+        +--------------------------+------------------------------------+
+        | author.name              |        256 characters              |
+        +--------------------------+------------------------------------+
+
+
+        Returns:
+            bool: True if the embed fits within Discord limits
+
+        Raises:
+            ValueError: If the embed is not valid and Discord would return an exception if sent
```

Comment:
Might be better worded as the following, although this could probably be better.
```suggestion
        Raises:
            ValueError: One or more of the embed attributes are too long.
```